src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php

```diff
@@ -4,33 +4,84 @@
 
 namespace Sylius\Bundle\PromotionBundle\Form\Type;
 
-use Sylius\Bundle\PromotionBundle\Form\Type\CatalogAction\PercentageDiscountActionConfigurationType;
 use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
 use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
 use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
 
 final class CatalogPromotionActionType extends AbstractResourceType
 {
-    private array $actionTypes;
+    private array $actionTypes = [];
 
-    public function __construct(string $dataClass, array $validationGroups = [], array $actionTypes = [])
-    {
+    private array $actionConfigurationTypes;
+
+    public function __construct(
+        string $dataClass,
+        array $validationGroups,
+        iterable $actionConfigurationTypes
+    ) {
         parent::__construct($dataClass, $validationGroups);
 
-        $this->actionTypes = $actionTypes;
+        foreach ($actionConfigurationTypes as $type => $formType) {
+            $this->actionConfigurationTypes[$type] = get_class($formType);
+            $this->actionTypes['sylius.form.catalog_promotion.action.' . $type] = $type;
+        }
+        ksort($this->actionTypes);
```

Comment:
Why do we sort these action types? Won't it be better to just use Symfony order?